torchviz/dot.py

```diff
@@ -6,68 +6,134 @@
     """ Produces Graphviz representation of PyTorch autograd graph.
 
     Blue nodes are the Variables that require grad, orange are Tensors
-    saved for backward in torch.autograd.Function
-
+    saved for backward in torch.autograd.Function. If output is a view,
+    there is a dotted edge between it and its base.
     Args:
         var: output Variable
         params: dict of (name, Variable) to add names to node that
             require grad (TODO: make optional)
+        show_attrs: whether to display the non-tensor attrs of the backward nodes
+        show_saved: whether to display saved variable nodes
     """
     if params is not None:
         assert all(isinstance(p, Variable) for p in params.values())
         param_map = {id(v): k for k, v in params.items()}
+    else:
+        param_map = {}
 
     node_attr = dict(style='filled',
                      shape='box',
                      align='left',
-                     fontsize='12',
+                     fontsize='10',
                      ranksep='0.1',
-                     height='0.2')
+                     height='0.2',
+                     fontname='monospace')
     dot = Digraph(node_attr=node_attr, graph_attr=dict(size="12,12"))
     seen = set()
 
     def size_to_str(size):
         return '(' + (', ').join(['%d' % v for v in size]) + ')'
 
-    output_nodes = (var.grad_fn,) if not isinstance(var, tuple) else tuple(v.grad_fn for v in var)
-
-    def add_nodes(var):
-        if var not in seen:
-            if torch.is_tensor(var):
-                # note: this used to show .saved_tensors in pytorch0.2, but stopped
-                # working as it was moved to ATen and Variable-Tensor merged
-                dot.node(str(id(var)), size_to_str(var.size()), fillcolor='orange')
-            elif hasattr(var, 'variable'):
-                u = var.variable
-                name = param_map[id(u)] if params is not None else ''
-                node_name = '%s\n %s' % (name, size_to_str(u.size()))
-                dot.node(str(id(var)), node_name, fillcolor='lightblue')
-            elif var in output_nodes:
-                dot.node(str(id(var)), str(type(var).__name__), fillcolor='darkolivegreen1')
-            else:
-                dot.node(str(id(var)), str(type(var).__name__))
-            seen.add(var)
-            if hasattr(var, 'next_functions'):
-                for u in var.next_functions:
-                    if u[0] is not None:
-                        dot.edge(str(id(u[0])), str(id(var)))
-                        add_nodes(u[0])
-            if hasattr(var, 'saved_tensors'):
-                for t in var.saved_tensors:
-                    dot.edge(str(id(t)), str(id(var)))
-                    add_nodes(t)
+    def get_var_name(var, name=None):
+        name = param_map[id(var)] if id(var) in param_map else ''
+        return '%s\n %s' % (name, size_to_str(var.size()))
+
+    def add_nodes(fn):
```

Comment:
this function used to handle inputs being a Tensor.
This is never the case anymore right?
Can you add an assert here maybe?